vault/vault.go

```diff
@@ -651,15 +659,30 @@ func (v *Vault) Copy(oldpath, newpath string, opts MoveCopyOpts) error {
 		t, err := v.ConstructSecrets(srcPath, TreeOpts{
 			FetchKeys:           true,
 			GetOnly:             true,
-			FetchAllVersions:    opts.Deep,
+			FetchAllVersions:    opts.Deep || srcVersion != 0,
 			GetDeletedVersions:  opts.Deep && opts.DeletedVersions,
-			AllowDeletedSecrets: opts.Deep,
+			AllowDeletedSecrets: opts.Deep || srcVersion != 0,
 		})
 
 		if err != nil {
 			return err
 		}
 
+		if len(t) == 0 {
+			// Prevent a panic
+			return fmt.Errorf("No valid secrets were found to be copied")
```

Comment:
this error should probably be normalized to whatever the normal not found is, something like `no secret was found at path 'blah'`